client_test.go

```diff
@@ -48,34 +47,31 @@ func TestNewClient(t *testing.T) {
 	//	assert.Equal(t, defaultBaseURL, c.BaseURL, )
 }
 
-func Test_client_CreateToken(t *testing.T) {
-	c := initClient(t)
-
-	var expire = time.Now().Add(time.Hour)
-	tt := []struct {
-		name   string
+func TestClient_CreateToken(t *testing.T) {
+	type args struct {
+		userId string
 		expire time.Time
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []byte
+		wantErr bool
 	}{
-		{"token without expire", time.Time{}},
-		{"token with expire", expire},
+		{"simple without expiration",  args{"tommaso", time.Time{}}, []byte("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoidG9tbWFzbyJ9.v-x-jt3ZnBXXbQ0GoWloIZtVnat2IE74U1a4Yuxd63M"), false},
+		{"simple with expiration",  args{"tommaso", time.Unix(1566941272, 0)}, []byte("eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE1NjY5NDEyNzIsInVzZXJfaWQiOiJ0b21tYXNvIn0.jF4ZbAIEuzS2jRH0uiu3HW9n0NHwT96QkzGlywcG9HU"), false},
 	}
-
-	for _, test := range tt {
-		test := test
-		t.Run(test.name, func(t *testing.T) {
-			token, err := c.CreateToken(testUsers[0].ID, test.expire)
-			mustNoError(t, err, "create token")
-
-			claims, err := jwt.HMACCheck(token, c.apiSecret)
-			mustNoError(t, err, "jwt check")
-
-			var expiresIn *jwt.NumericTime
-			if !test.expire.IsZero() {
-				expiresIn = jwt.NewNumericTime(test.expire)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c, _ := NewClient("key", []byte("secret"))
+			got, err := c.CreateToken(tt.args.userId, tt.args.expire)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("createToken() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
It's simple but could be even shorter by `testify`. Since it is in vendor, why not use it?

Reply:
The plan is to drop testify. It was left in there to save time during the rushed launch.

Reply:
Sounds good to me then.

For my own learning, what is the reason to drop?

Reply:
What I dislike most is the unreadable errors it produces. To make things even worse, the framework makes the description optional. Spending 30 seconds on a good Errorf description goes a long way.

Reply:
To add from my experience, if diffing long strings and there is a mismatch, it shows no diff. 

It's generally very confusing because at the exact same place, sometimes you see an error and sometimes it's eaten according to the inputs.